### PR TITLE
[Otel exporter] API updates

### PR DESCRIPTION
**Pull request by annatisch**

### CODE REVIEW

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_connection_string_parser.py [hunk 1]
```diff
@@ -28,9 +28,12 @@ class ConnectionStringParser:
 
     def __init__(
         self,
-        connection_string: str = None
+        connection_string: typing.Optional[str] = None,
+        *,
+        instrumentation_key: typing.Optional[str] = None,
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_connection_string_parser.py:33]: Why are you adding this?

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_export/_base.py [hunk 2]
```diff
@@ -62,22 +66,35 @@ class ExportResult(Enum):
 class BaseExporter:
     """Azure Monitor base exporter for OpenTelemetry."""
 
-    def __init__(self, **kwargs: Any) -> None:
+    def __init__(
+        self,
+        *,
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_export/_base.py:71]: If the args are never used, do we really want to put *, here?

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_export/_base.py [hunk 3]
```diff
@@ -62,22 +66,35 @@ class ExportResult(Enum):
...
+        :paramtype credential: ~azure.core.credentials.TokenCredential
+        :keyword bool disable_offline_storage: Determines whether to disable storing failed telemetry records
+         for retry. Defaults to `False`.
+        :keyword str storage_directory: Storage path in which to store retry files. Defaults
+         to `<tempfile.gettempdir()>/opentelemetry-python-<your-instrumentation-key>`.
         :rtype: None
         """
-        parsed_connection_string = ConnectionStringParser(kwargs.get('connection_string'))
+        parsed_connection_string = ConnectionStringParser(
+            kwargs.get('_conn_str'),
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_export/_base.py:90]: This is a pretty major change. We would have to update all our our documentation and conflict with the other languages. Why is this important?

File: sdk/monitor/azure-monitor-opentelemetry-exporter/tests/trace/test_trace.py [hunk 4]
```diff
@@ -58,7 +58,7 @@ def tearDownClass(cls):
     def test_constructor(self):
         """Test the constructor."""
         exporter = AzureMonitorTraceExporter(
-            connection_string="InstrumentationKey=4321abcd-5678-4efa-8abc-1234567890ab",
+            instrumentation_key="4321abcd-5678-4efa-8abc-1234567890ab",
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/tests/trace/test_trace.py:61]: We instantiate with connection string, not ikey. Why is this change here?

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_export/_base.py (hunk 3 above)
Comment (jeremydvoss) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_export/_base.py:90]: Not going to mark every spot where this is used for the sake of keeping things clear.

Comment (annatisch) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_export/_base.py:90]: We should only support the passing of connection string via one way.
The intention of the addition of the `from_connection_string` classmethod, is that this would become the only way to do so.
Normally I would advocate for a more complete refactor in that parsing of the connection string would happen in the classmethod itself, not the main constructor, however given the complexities of your parser that combines with the env var parsing, I decided the simplest approach for now was just to make the constructor param private/internal-only.

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_connection_string_parser.py (hunk 1 above)
Comment (annatisch) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_connection_string_parser.py:33]: I added this based on the comments further down that implied this was a supported scenario (2nd in the priority heirarchy), but was missing. I can revert this and amend the comment if that is inaccurate?

File: sdk/monitor/azure-monitor-opentelemetry-exporter/tests/trace/test_trace.py (hunk 4 above)
Comment (annatisch) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/tests/trace/test_trace.py:61]: I only did this in relation to your comment earlier about the addition of `instrumentation_key` to the connection string parser. I will update these tests depending on the outcome of that thread. However if we chose to only instantiate via connection string, this test becomes redundant, as we should only support a single way to instantiate with connection string, which is already tested.

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_export/_base.py (hunk 2 above)
Comment (annatisch) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_export/_base.py:71]: This behaviour is identical to the existing `**kwargs` syntax, just declares the contents of kwargs more specifically to support type hinting and a better IDE/intellisense experience.

Comment (jeremydvoss) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_export/_base.py:71]: Got it.

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_export/_base.py (hunk 3 above)
Comment (jeremydvoss) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_export/_base.py:90]: I don't understand. We only added from_connection_string because azure sdk previously said to. If we removed it, does that mean we could still have connection string as an argument in the exporter constructor?
